### PR TITLE
tor s - octos - api-muncher

### PR DESCRIPTION
**Pull request by torshimizu**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
How did you go about exploring the Edamam API, how did you try querying the API? |  I read the docs, tried Postman, tried things in atom, realized I didn't have what I needed, reread the docs then retried things in Postman. 
Describe your API Wrapper.  How did you decide on the methods you created? | I had a method to get a list of recipes and to get one recipe. I decided on two methods because of the nature of Edamam. Edamam takes a different parameter for a single recipe search, so I needed a different method to account for that.
Describe an edge case or failure case test you wrote for your API Wrapper. | If Edemam sent back  a 404, I had to rescue the error. I tried my best to test for this, but it was difficult to know how to get Edamam to send an error. I also added a test for Edemam not sending back any entries.
Explain how VCR aids in testing an API. | VCR lets you save queries for indexing so that you do not use up limited requests when testing.
What is the Heroku URL of your deployed application? | https://octo-muncher.herokuapp.com/
Provide a link to the Trello board you used | https://trello.com/b/gAcjE6d6/muncher

### CODE REVIEW

**Comment by torshimizu** [comment]
If it's not too much to ask, could mine please, please be looked at later? I really didn't get to finish/polish up things and I wasn't the most comfortable submitting what I had. 

Please and thank you!

File: test/controllers/favorites_controller_test.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+      recipe_data = {
+        name: 'Teriyaki Chicken',
+        uri: 'www.goodfood.com',
+        image: 'www.goodfood.com/image/1'
+      }
+
+      post favorites_path, params: recipe_data
+
+      must_respond_with :success
```
Comment (CheezItMan) [inline test/controllers/favorites_controller_test.rb:20]: This should be a redirect as it redirects back.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    <title><%= content_for?(:title) ? yield(:title) : "Octo-Muncher" %></title>
+
+    <script src="https://developer.edamam.com/attribution/badge.js"></script>
+
+    <%= stylesheet_link_tag    "application" %>
+    <%= javascript_include_tag "application", 'data-turbolinks-track' => true %>
+    <%= csrf_meta_tags %>
+  </head>
+
+  <body class='grid-y'>
```
Comment (CheezItMan) [inline app/views/layouts/application.html.erb:16]: Foundation xy-grid?  neat

File: test/controllers/recipes_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    it 'wont add a search term that produces no results' do
+      VCR.use_cassette('recipes') do
+        get recipes_path, params: {query: 'boeing'}
+
+        session[:recent_searches].wont_include 'boeing'
+      end
+    end
+    # =======================
+
+    it 'must return ok for no recipes' do
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:39]: Maybe also test the flash notices?

File: test/controllers/recipes_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      VCR.use_cassette('recipes') do
+        get recipes_path, params: {query: 'some-non-indexed-dish'}
+
+        must_respond_with :ok
+      end
+    end
+
+
+    it 'redirects to root when an error is received' do
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:48]: This can be removed

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
+class SessionsController < ApplicationController
+
+  before_action :check_if_logged_in, only: [:create]
+
+  def create
+    auth_hash = request.env['omniauth.auth']
+
+    if auth_hash['uid']
+      user = User.build_from_google(auth_hash)
```
Comment (CheezItMan) [inline app/controllers/sessions_controller.rb:9]: 🥇

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Core Requirements** | 
Git hygiene | Good number of commits and good commit messages
Comprehension questions | Check, I'm happy to look at this again if you get additional features done, but you already have the requirements.  
**General** | 
Rails fundamentals (RESTful routing, use of named paths) | Your show action isn't RESTful, to follow the pattern it should be `/recipes/:id`, 
Semantic HTML | Check
Errors are reported to the user | Check
API Wrapper to handle the API requests |   Check 
Controller testing | Check
Lib testing | Check
Search Functionality | Check
List Functionality | Check
Show individual item functionality (link to original recipe opens in new tab) | Check
**Styling** | 
Responsive layout | Check, both in search results and show page
List View shows 10 items at a time/pagination | Check
The app is styled to create an attractive user interface | Nicely done 
**API Features** | 
The App attributes Edaman | Check
The VCR casettes do not contain the API key | 
**External Resources** | Check
Link to Trello Board | Check, although I don't have access
Link to deployed app on Heroku | Nicely done
**Overall** | Nice work on the Google OAuth.  If you get the chance show it to Zheng in the Ampers as she's working on the same and did Amazon OAuth.  You hit all the learning goals and some nice extras with favorites and Google OAuth.  Excellent job!
